File: src/test/java/connectivity/JdsDbMySqlImplementation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package connectivity;

import io.github.subiyacryolite.jds.JdsDbMySql;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Properties;

/**
 * @author indana
 */
public class JdsDbMySqlImplementation extends JdsDbMySql {

    @Override
    public Connection getConnection() throws ClassNotFoundException, SQLException {
        Class.forName("com.mysql.cj.jdbc.Driver");
        Properties properties = new Properties();
        properties.put("user", "root");
        properties.put("password", "p@nkP#55W0rd");
        //properties.put("password", "");
        properties.put("autoReconnect", "true");
        properties.put("allowMultiQueries", "false");
        properties.put("useSSL", "false");
        properties.put("rewriteBatchedStatements", "true");
        properties.put("continueBatchOnError", "true");
        return DriverManager.getConnection("jdbc:mysql://localhost:3306/jds", properties);
    }
}

File: src/test/kotlin/connectivity/JdsDbMySqlImplementation.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package connectivity

import io.github.subiyacryolite.jds.JdsDbMySql
import java.io.File
import java.io.FileInputStream
import java.sql.Connection
import java.sql.DriverManager
import java.util.*

class JdsDbMySqlImplementation : JdsDbMySql() {

    private val properties: Properties = Properties()
    private val connectionString: String

    init {
        FileInputStream(File("db.mysql.properties")).use { properties.load(it) }
        connectionString = "jdbc:mysql://${properties["dbUrl"]}:${properties["dbPort"]}/${properties["dbName"]}"
    }

    override val connection: Connection
        get () {
            Class.forName("com.mysql.cj.jdbc.Driver")
            return DriverManager.getConnection(connectionString, properties)
        }
}